### PR TITLE
cloudcore  watch pvc and pv 

### LINKED ISSUE #1899: delete pvc and pv that edgecore sqlite cannot delete information
**What happened**:
edgecore  sqlite3 can not clear pv and pvc of deletion information beacaus  cloudcore do not send deletion message  for  edgehub 
**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:
ZIOKVDBBA3%W@B7UP4YF6ZH

**Environment**:
- KubeEdge version(e.g. `cloudcore/edgecore --version`):  
edgecore1.3.0
**CloudSide Environment**:
- Hardware configuration (e.g. `lscpu`):
- OS (e.g. `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Go version (e.g. `go version`): 
- Others:

**EdgeSide Environment**:
- edgecore version (e.g. `edgecore --version`):
- Hardware configuration (e.g. `lscpu`):
- OS (e.g. `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Go version (e.g. `go version`): 
- Others:

### PR DESCRIPTION
**Pull request by guojianyu**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:

/kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind test
> /kind failing-test
> /kind feature

**What this PR does / why we need it**:
this bug make pvc name cannot be reused,local pv directory name is first pv name all the time.
provisioner local pv solved this bug can support
**Which issue(s) this PR fixes**:

Fixes #1899

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by guojianyu** [comment]
pass-test

**Comment by fisherxu** [comment]
@guojianyu Thanks for the enhancement, and does the edge side need any modify?

**Comment by guojianyu** [comment]
> @guojianyu Thanks for the enhancement, and does the edge side need any modification?

edgecore does not need to change,it has standard processes.

File: cloud/pkg/edgecontroller/controller/downstream.go [hunk 1]
```diff
@@ -438,6 +442,117 @@ func (dc *DownstreamController) syncEndpoints() {
...
+				operation = model.UpdateOperation
+			case watch.Deleted:
+				dc.lc.DeletePersistentvolumeclaim(*pvc)
+				operation = model.DeleteOperation
+			default:
+				// unsupported operation, no need to send to any node
+				klog.Warningf("Persistentvolumclaim event type: %s unsupported", e.Type)
+				continue
+			}
+			// send to all nodes
```
Comment (fisherxu) [inline cloud/pkg/edgecontroller/controller/downstream.go:530]: Do we need to send to all nodes? Just sends to the node in localCache.

File: cloud/pkg/edgecontroller/controller/downstream.go [hunk 2]
```diff
@@ -438,6 +442,117 @@ func (dc *DownstreamController) syncEndpoints() {
...
+				operation = model.UpdateOperation
+			case watch.Deleted:
+				dc.lc.DeletePersistentVolume(*pv)
+				operation = model.DeleteOperation
+			default:
+				// unsupported operation, no need to send to any node
+				klog.Warningf("persistentVolume event type: %s unsupported", e.Type)
+				continue
+			}
+			// send to all nodes
```
Comment (fisherxu) [inline cloud/pkg/edgecontroller/controller/downstream.go:474]: ditto

File: cloud/pkg/edgecontroller/controller/downstream.go (hunk 1 above)
Comment (guojianyu) [reply cloud/pkg/edgecontroller/controller/downstream.go]: > Do we need to send to all nodes? Just sends to the node in localCache.

get it,I have updated it. 
pvcmap and pvmap key is namespace/name ,value is nodename list.
 add pod and update pod sync pvcmap to cache. 
 add pvc and update pvc  sync pvmap to cache.

File: cloud/pkg/edgecontroller/manager/persistentvolumeclaim.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+// Events return the channel save events from watch Persistentvolumeclaim change
+func (sm *PersistentvolumeclaimManager) Events() chan watch.Event {
+	return sm.events
+}
+
+// NewPersistentvolumeclaimManager create PersistentvolumeclaimManager by kube clientset and namespace
+func NewPersistentvolumeclaimManager(kubeClient *kubernetes.Clientset, namespace string) (*PersistentvolumeclaimManager, error) {
+	lw := cache.NewListWatchFromClient(kubeClient.CoreV1().RESTClient(), "persistentvolumes", namespace, fields.Everything())
```
Comment (fisherxu) [inline cloud/pkg/edgecontroller/manager/persistentvolumeclaim.go:24]: The resource seems not correct here

File: cloud/pkg/edgecontroller/manager/persistentvolumeclaim.go [hunk 4]
```diff
@@ -0,0 +1,33 @@
+package manager
+
+import (
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	//"github.com/kubeedge/kubeedge/cloud/pkg/edgecontroller/config"
```
Comment (fisherxu) [inline cloud/pkg/edgecontroller/manager/persistentvolumeclaim.go:9]: delete the unused import

File: cloud/pkg/edgecontroller/manager/location.go [hunk 5]
```diff
@@ -243,3 +288,25 @@ func (lc *LocationCache) GetAllEndpoints() []v1.Endpoints {
 	})
 	return endpoints
 }
+
+// AddOrUpdatePersistentVolume in cache
+func (lc *LocationCache) AddOrUpdatePersistentVolume(namespace, name string, nodes []string) {
```
Comment (fisherxu) [inline cloud/pkg/edgecontroller/manager/location.go:293]: Where call this func?

File: cloud/pkg/edgecontroller/controller/downstream.go [hunk 6]
```diff
@@ -439,6 +443,110 @@ func (dc *DownstreamController) syncEndpoints() {
...
+				operation = model.UpdateOperation
+			case watch.Deleted:
+				operation = model.DeleteOperation
+			default:
+				// unsupported operation, no need to send to any node
+				klog.Warningf("persistentVolume event type: %s unsupported", e.Type)
+				continue
+			}
+
+			nodes := dc.lc.PersistentvolumNodes(pv.Namespace, pv.Name)
```
Comment (fisherxu) [inline cloud/pkg/edgecontroller/controller/downstream.go:473]: it seems the `PersistentvolumNodes` has not been populated.

File: cloud/pkg/edgecontroller/manager/persistentvolumeclaim.go (hunk 3 above)
Comment (guojianyu) [reply cloud/pkg/edgecontroller/manager/persistentvolumeclaim.go:24]: > The resource seems not correct here

get it ,local latest forgot to update.

File: cloud/pkg/edgecontroller/manager/persistentvolumeclaim.go (hunk 4 above)
Comment (guojianyu) [reply cloud/pkg/edgecontroller/manager/persistentvolumeclaim.go:9]: > delete the unused import

get it

File: cloud/pkg/edgecontroller/controller/downstream.go (hunk 6 above)
Comment (guojianyu) [reply cloud/pkg/edgecontroller/controller/downstream.go:473]: > it seems the `PersistentvolumNodes` has not been populated.

location.go have content.

**Comment by guojianyu** [comment]
pass-test

**Pull request closed without merging**